pkgs/by-name/wh/whisper-cpp/package.nix

```diff
@@ -0,0 +1,185 @@
+    # have stopped using lndir: https://github.com/NixOS/nixpkgs/issues/271792
+    cuda_cudart
+    libcublas
+  ];
+
+  rocmBuildInputs = with rocmPackages; [
+    clr
+    hipblas
+    rocblas
+  ];
+
+  vulkanBuildInputs = [
+    shaderc
+    vulkan-headers
+    vulkan-loader
+  ];
+
+in
+effectiveStdenv.mkDerivation (finalAttrs: {
+  pname = "whisper-cpp";
+  version = "1.7.2";
+
+  src = fetchFromGitHub {
+    owner = "ggerganov";
+    repo = "whisper.cpp";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-y30ZccpF3SCdRGa+P3ddF1tT1KnvlI4Fexx81wZxfTk=";
+  };
+
+  # The upstream download script tries to download the models to the
+  # directory of the script, which is not writable due to being
+  # inside the nix store. This patch changes the script to download
+  # the models to the current directory of where it is being run from.
+  patches = [ ./download-models.patch ];
+
+  nativeBuildInputs =
+    [
+      cmake
+      ninja
+      which
+      makeWrapper
+    ]
+    ++ lib.optionals cudaSupport [
+      cudaPackages.cuda_nvcc
+      autoAddDriverRunpath
+    ];
+
+  buildInputs =
+    optional withSDL SDL2
+    ++ optionals effectiveStdenv.hostPlatform.isDarwin darwinBuildInputs
+    ++ optionals cudaSupport cudaBuildInputs
+    ++ optionals rocmSupport rocmBuildInputs
+    ++ optionals vulkanSupport vulkanBuildInputs;
+
+  cmakeFlags =
+    [
+      "-DCMAKE_SKIP_BUILD_RPATH=ON" # the project doesn't install the main binaries in their installation target
+      (cmakeBool "GGML_CUDA" cudaSupport)
+      (cmakeBool "GGML_HIPBLAS" rocmSupport)
+      (cmakeBool "GGML_VULKAN" vulkanSupport)
+      (cmakeBool "WHISPER_SDL2" withSDL)
+      (cmakeBool "GGML_LTO" true)
+      (cmakeBool "GGML_NATIVE" false)
+      (cmakeBool "BUILD_SHARED_LIBS" (!effectiveStdenv.hostPlatform.isStatic))
+    ]
+    ++ optionals (effectiveStdenv.hostPlatform.isx86 && !effectiveStdenv.hostPlatform.isStatic) [
+      (cmakeBool "GGML_BACKEND_DL" true)
+      (cmakeBool "GGML_CPU_ALL_VARIANTS" true)
+    ]
+    ++ optionals effectiveStdenv.hostPlatform.isDarwin [
+      (cmakeBool "GGML_METAL" metalSupport)
+      (cmakeBool "GGML_METAL_EMBED_LIBRARY" metalSupport)
+      (cmakeBool "WHISPER_COREML" true)
+      (cmakeBool "WHISPER_COREML_ALLOW_FALLBACK" true)
+    ]
+    ++ optionals cudaSupport [
+      (cmakeFeature "CMAKE_CUDA_ARCHITECTURES" cudaPackages.flags.cmakeCudaArchitecturesString)
+    ]
+    ++ optionals rocmSupport [
+      (cmakeFeature "CMAKE_C_COMPILER" "hipcc")
+      (cmakeFeature "CMAKE_CXX_COMPILER" "hipcc")
+
+      # Build all targets supported by rocBLAS. When updating search for TARGET_LIST_ROCM
+      # in https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/CMakeLists.txt
+      # and select the line that matches the current nixpkgs version of rocBLAS.
+      "-DAMDGPU_TARGETS=${lib.concatStringsSep ";" rocmPackages.clr.gpuTargets}"
```

Comment:
Could this be moved to args to be able to override it? See https://github.com/ggerganov/llama.cpp/pull/10794

Reply:
Of course. I'll add this.
I also think the ROCm version should be tagged as big parallel.

Reply:
Done